plugins/plugin-bitbucket/codenvy-plugin-bitbucket-ext-bitbucket-server/src/main/java/org/eclipse/che/ide/ext/bitbucket/server/rest/BitbucketService.java

```diff
@@ -130,10 +131,21 @@ public BitbucketPullRequest openPullRequest(@PathParam("owner") String owner,
                                                 BitbucketPullRequest pullRequest) throws IOException,
                                                                                          BitbucketException,
                                                                                          ServerException {
-
         return bitbucketConnection.openPullRequest(owner, repositorySlug, pullRequest);
     }
 
+    @PUT
+    @Path("repositories/{owner}/{repositorySlug}/pullrequests")
+    @Consumes(APPLICATION_JSON)
+    @Produces(APPLICATION_JSON)
+    public BitbucketPullRequest updatePullRequest(@PathParam("owner") String owner,
+                                                  @PathParam("repositorySlug") String repositorySlug,
+                                                  BitbucketPullRequest pullRequest) throws IOException,
+                                                                                           BitbucketException,
+                                                                                           ServerException {
+        return bitbucketConnection.updatePullRequest(owner, repositorySlug, pullRequest);
+    }
+
```

Comment:
1. Please take a look https://github.com/codenvy/codenvy/pull/2000/files#diff-5e9c059cfdf4fce8aaf00131f0c424adR155 As far as I remember we've started use `vcs` instead of `git` for service name. It can be a problem with showing such keys by IDE. Please check that IDE shows new generated keys.
2. As far as I understand this https://github.com/codenvy/codenvy/pull/2000/files#diff-5e9c059cfdf4fce8aaf00131f0c424adR152 won't work for bitbucket server that can have different host. Am I correct?